azure-iot-device/azure/iot/device/common/pipeline/pipeline_thread.py

```diff
@@ -166,6 +166,15 @@ def invoke_on_callback_thread_nowait(func):
     return _invoke_on_executor_thread(func=func, thread_name="callback", block=False)
 
 
+def invoke_on_http_thread_nowait(func):
+    """
+    Run the decorated function on the callback thread, but don't wait for it to complete
+    """
+    # TODO: Refactor this since this is not in the pipeline thread anymore, so we need to pull this into common.
```

Comment:
># TODO: Refactor this since this is not in the pipeline thread anymore, so we need to pull this into common. [](start = 4, length = 108)

We should see if the code can better with manual threading.  Same model, just without the decorator magic.

Reply:
I can work on that after this PR is finished. It's a good idea but maybe save for later.

---
In reply to: [352774880](https://github.com/Azure/azure-iot-sdk-python/pull/403#discussion_r352774880) [](ancestors = 352774880)